go/vt/mysqlctl/mysqld.go

```diff
@@ -676,6 +676,14 @@ func (mysqld *Mysqld) Init(ctx context.Context, cnf *Mycnf, initDBSQLFile string
 	return nil
 }
 
+func readMysqldErrorLog(fileName string) string {
+	b, err := ioutil.ReadFile(fileName)
```

Comment:
We can't always assume it's possible to read the error log. For example, in our operator we redirect the mysqld error log to stderr so that it gets funneled through the normal Kubernetes container log flow. We do this by setting the error log path to (a symlink to) `/dev/stderr`, so trying to read from that path won't work as expected. It's likely that the failure mode will be "ReadFile() blocks forever because EOF never occurs", so even ignoring the error doesn't handle this case.

Reply:
As discussed on Slack, I have changed it to check it is a regular file, and only proceed if it is. The last 4KiB of the file are read.